### ISSUE TITLE
for the output

### ISSUE THREAD

**Issue by HA-sala** [gaolinorange/SDMNET_stamp#2]
Hello, I have imitated the labels of your dataset and made some models as my datasets  (all models have different structures).I used my dataset to train the SDM_NET, and Shape Reconstruction can perfectly reconstruct the model in the data set, But Shape Generation and Shape Interpolation can only output the same model that comes from the dataset, Whether it's retraining or re-running.

It could be a model collapse, and Would you please  give me some advice? Thank you.

**Comment by tommaoer** [comment]
Sorry for the delayed reply. I had a ddl some times ago. For the problem, I think you should adjust the weight of reconstruction and KL-divergence to learn a smooth latent space for generation.
If you have any difficulties, please feel free to contact me.

**Comment by HA-sala** [comment]
谢谢，我去试试

**Comment by chengzg** [comment]
@HA-sala ,

Have you managed to get some output with your own dataset? 
I am also interested to try it with my own dataset. Can you give a brief steps on how to prepare my own data and how to start the training process with my own data? Thanks very much for your help in advance.

Thanks & Regards,
Spencer

**Comment by HA-sala** [comment]
@chengzg ,

The process maybe a little complicated.

First, make models with NX or other.

Then save the established models according to the style of the datasets provided by the author (note to open each file with Notepad to see the detailed format).

 Finally import the saved model file into MATLAB, and use the Pipeline script to make the network training set.

Of course, you may need to understand the matlab scripts provided by the author, well.

**Comment by chengzg** [comment]
@HA-sala 
@shinxg 

Thanks very much for your reply.

Regarding to 3D model, can you specifically point out what kind of style should i refer to, for example chair data. Also can you point out which file do you refer in current Repo (open in Notepad to see the detailed format)?  Do you refer to the chair.obj and plane.obj?

Also i cannot find where the *_vaeid.dat is generated, do you know where is it being generated? Thanks a lot

Thanks & Regards,
Spencer

**Comment by HA-sala** [comment]
@chengzg 
for 3D model:
      for example car  data:
      you use Computer Notepad in The computer to open body.obj , you can see 

"# OBJ File Generated by Meshlab" 

in the top , This passage explains where the file was generated. 
In the same way, open model_normalized.obj, you can see 
"
# Alias OBJ Model File
# Exported from SketchUp, (c) 2000-2012 Trimble Navigation Limited
# File units = inches
"
Then, 
for where the* _vaefeature.mat is generated
      In, matlab     , find the file --- Pipeline.m, you can see some functions

     GetTransformedCube(categarydir, postfix, cate);     %%    from box50 file to get    transformed_cube_*   .obj
     regist(box_dir, cate)       %% for geting  * _reg.obj
    SupportAnalysisScript(box_dir, cate);%%for geting  code.mat  
%
%     Contents of code.mat
%     column 1     |  column 2->8   |  column 9->15   | column 16->18  |      column  19                 | column  20->23 
%      Existence   |   Supporting    |     Supported     |       Center        | Existence of Symmetry        | Parameter of Symmetry Plane
%

for where the  *_vaeid.dat is generated
      in pyhon code ,  utils.py, a function --- def spilt_dataset(num, percent_to_train, name="id.dat"):
      *_vaeid.dat   is only used here for spilting dataset. I can't find this file either, but  this function  gives a answer:
"
if os.path.isfile(name):
     ****
else:
    ****
"

**Comment by HA-sala** [comment]
for 3D model:
for example car data:
you use Computer Notepad in The computer to open body.obj . in the top,  you can see

"
# OBJ File Generated by Meshlab
"

 , This passage explains where the file was generated.
In the same way, open model_normalized.obj, you can see
"
Alias OBJ Model File
Exported from SketchUp, (c) 2000-2012 Trimble Navigation Limited
File units = inches
"
Then,
for where the * _vaefeature.mat is generated
In, matlab , find the file --- Pipeline.m, you can see some functions
"
 GetTransformedCube(categarydir, postfix, cate);     %%    from box50 file to get    transformed_cube_*   .obj
 regist(box_dir, cate)       %% for geting  * _reg.obj
 SupportAnalysisScript(box_dir, cate);%%for geting  code.mat  
%
% Contents of code.mat
% column 1 | column 2->8 | column 9->15 | column 16->18 | column 19 | column 20->23
% Existence | Supporting | Supported | Center | Existence of Symmetry | Parameter of Symmetry Plane
%.

generate_data(box_dir,vaedir,cate,partname,'');

%          for geting the all of vaenew50  file
%           and   geting   *_vaefeature.mat  
"
for where the *_vaeid.dat is generated
in pyhon code , utils.py, a function --- def spilt_dataset(num, percent_to_train, name="id.dat"):
*_vaeid.dat is only used here for spilting dataset. I can't find this file either, but this function gives a answer:
"
if os.path.isfile(name):
****
else:  
****
"

**Comment by chengzg** [comment]
@HA-sala 

Thank you so much for your help.  Really really appreciated.

I followed your advice and try to prepare one dataset with one simple obj file(chair) by running the pipeline.m file. However, i cant get it running. 

- Would you mind share some information about the hierarchy of your matlab folder containing your own training dataset?
- Also how do you prepare the segmentation data for the obj file. I can do manual annotation but do not know how to store it?
- How is the result of testing with your own data?
- Is your repo is the same as the current repo? I do not have the 'model_normalized.obj' file

Thanks & Regards,
Spencer

**Comment by chengzg** [comment]
@shinxg 
@tommaoer 

I really would like to try the network with my own dataset to see how the network works. 

- Is the current published repo able to train with our own dataset?

- If yes, can you help provide some guidelines on how should i prepare my dataset for the SDM-Net Training? When i step through the pipeline.m, i feel there are a lot of information missing such as files or folder hierarchy not in the repo. Thanks a lot.

Thanks & Regards,
Spencer

**Comment by tommaoer** [comment]
@chengzg You can try the other released version on our project page (http://geometrylearning.com/sdm-net/)， in the repo, we provide the detailed preprocess scripts for the shapenet data. If you want to try on your own dataset, you should change the format of your data which is the same as the shapenet, and try again.

If you have some further questions, please feel free to contact us.

**Comment by chengzg** [comment]
@tommaoer 
Thanks very much. That's what i am missing. So actually the repo works on Windows Environment other than Linux since i see there are some mexw64 file? Or just matlab related code works on Windows?

Thanks & Regards,
Spencer

**Comment by tommaoer** [comment]
@chengzg Yes, the preprocessing script is implemented by Matlab on windows os. I am very sorry that it is not very convenient for some Linux users. But, furthermore, we will be willing to give the Linux-friendly preprocessing script.

If you need some help, please contact me.

**Comment by chengzg** [comment]
@tommaoer 

Thanks very much. I am able to generate the %category%_vaefeature file. I wonder is  "%args.model% + '_vaeid.dat'"(chair_vaeid.dat) file required for the training? If needed, how should I prepare the file?

Thanks & Regards,
Spencer
